packages/expo-ui/src/jetpack-compose/Shape/index.tsx

```diff
@@ -0,0 +1,71 @@
+import { requireNativeView } from 'expo';
+import { ViewStyle } from 'react-native';
+
+export type ShapeProps = {
+  /**
+   * Corner rounding percentage. Multiplied by the shorter dimension of the view to produce pixel values.
+   * @default 0.0
+   */
+  cornerRounding?: number;
+  /**
+   * Number between `0.0` and `1.0` that determines how much each line between vertices is "smoothed".
+   * @default 0.0
+   */
+  smoothing?: number;
+  /**
+   * Number of vertices. For `'POLYGON'` it must be at least `3.0`. For `'STAR'` and `'PILL_STAR'` it is a number of vertices for each of two radii (A 5-pointed star has 10 vertices.)
+   * @default 6.0
+   */
+  verticesCount?: number;
+  /**
+   * Inner radius of star-related shapes (`'STAR'` and `'PILL_STAR'`). Multiplied by the shorter dimension of the view to produce pixel values.
+   * @default 1.0
+   */
+  innerRadius?: number;
+  /**
+   * Radius of the circular shape. Multiplied by the shorter dimension of the view to produce pixel values.
+   * @default 1.0
+   */
+  radius?: number;
+  /** Style of the component */
+  style?: ViewStyle;
+  /** Color of the shape */
+  color?: string;
+};
+
+const ShapeNativeView: React.ComponentType<any> = requireNativeView('ExpoUI', 'ShapeView');
+
+function Star(props: ShapeProps) {
+  return <ShapeNativeView {...props} style={props.style} type="STAR" />;
+}
+
+function PillStar(props: ShapeProps) {
+  return <ShapeNativeView {...props} style={props.style} type="PILL_STAR" />;
+}
+
+function Pill(props: Pick<ShapeProps, 'smoothing' | 'style' | 'color'>) {
+  return <ShapeNativeView {...props} style={props.style} type="PILL" />;
+}
+
+function Circle(props: Pick<ShapeProps, 'smoothing' | 'verticesCount' | 'style' | 'color'>) {
```

Comment:
I'm pretty sure that the native circle is not using `smoothing` but it does use `redius` 
```suggestion
function Circle(props: Pick<ShapeProps, 'radius' | 'verticesCount' | 'style' | 'color'>) {
```